src/content/post/2025-05-05-kuzu-v-0.10.0.md

```diff
@@ -0,0 +1,300 @@
+| soc-sign-bitcoin-otc | 5K | 35K | 1MB |
+| soc-LiveJournal1 | 4.8M | 68M | 1.1GB |
+| datagen-sf10k | 100M | 9.4B | 294.3GB |
+
+We report the end-to-end runtimes on a machine with 2xAMD EPYC 9J14 CPUs and 768GB RAM, using 4, 16 and
+64 threads for execution.
+
+#### soc-sign-bitcoin-otc
+
+| Algorithm | 4 threads | 16 threads | 64 threads |
+|-----------|-----------|------------|------------|
+| WCC |  12.2ms | 15.7ms | 20.0ms |
+| SCC | 39.3ms | 46.7ms | 49.6ms |
+| SCC-ko | 15.5ms | 38.1ms | 34.5ms |
+| PageRank | 50.1ms | 51.0ms | 49.6ms |
+| K-Core | 181.3ms | 207.8ms | 213.0ms |
+| Louvain | 123.0ms | 137.2ms | 143.7ms |
+| WSP | 16.4ms | 16.8ms | 19.2ms |
+
+#### soc-LiveJournal1
+
+| Algorithm | 4 threads | 16 threads | 64 threads |
+|-----------|-----------|------------|------------|
+| WCC | 3.6s  | 0.9s | 0.3s |
+| SCC | 9.6s | 2.6s | 0.9s |
+| SCC-ko | 8.3s | 8.2s | 8.2s |
+| PageRank | 19.9s | 6.9s | 5.1s |
+| K-Core | 46.7s | 13.3s | 9.2s |
+| Louvain | 102.2s | 42.7s | 21.2s |
+| WSP | 13.6s | 3.8s | 1.1s |
+
+#### datagen-sf10k
+
+[^1]: Louvain algorithm requires more than 256GB of memory on datagen-sf10k so we omit Louvain in this dataset.
+
+| Algorithm | 4 threads | 16 threads | 64 threads |
+|-----------|-----------|------------|------------|
+| WCC | 201.4s | 54.3s | 19.7s |
+| SCC | >30min | >30min | 849.3s |
+| SCC-ko | 273.5s | 271.2s | 280.0s |
+| PageRank | 992.5s| 299.4s | 146.5s |
+| K-Core | 585.6s | 151.8s | 52.7s |
+| WSP | 4.8s | 2.5s | 0.7s |
+
+
+Above benchmarks show our algorithms are able to run on billion scale graph and scales well as number of threads increases.
+
+### Using graph algorithms
+
+All algorithms, except for that [(all) weighted shortest paths](http://docs.kuzudb.com/cypher/query-clauses/match#shortest-path) which is provided as built-in, are available via the [`algo` extension](http://docs.kuzudb.com/extensions/algos). 
+Let's look at example usage of graph algorithms using the well-known LDBC SNB dataset.
+The example beloew finds top 10 influencers based on the `(person)-[:knows]->(person)` subgraph and counts how many posts they have published.
+
+```cypher
+// Install graph algorthm extension
+INSTALL algo;
+LOAD algo;
+
+// Create a projected graph with person and knows tables.
+CALL project_graph(
+    'KnowsGraph', 
+    ['person'], 
+    ['knows']
+);
+
+// Run page rank on projected graph to find the top 10 influencers.
+// Then traverse `postHasCreator` to extract all posts
+CALL page_rank('KnowsGraph')
+WITH node AS person, rank
+ORDER BY rank DESC
+LIMIT 10
+MATCH (person)<-[:postHasCreator]-(post:Post)
+RETURN person, COUNT(*);
+```
+
+We first install and load the `algo` extension. Then, we create a projected graph from the `person` and `knows` tables,
+on which the algorithm is executed. The projected graph is a subgraph that contains only the nodes and
+relationships specified. It is evaluated only when the algorithm is executed. Kuzu does not materialize projected
```

Comment:
Do we want to include technical details of how projected graph is evaluated in the post? I would just point to doc.

Reply:
I want to include it because from a reader's perspective, it's not immediately clear why a projected graph is necessary. Actually, we should drop a brief line what the purpose of the projected graph is. For an uninitiated user, they might expect that the graph algorithm just runs on the full graph, without this intermediate step. We shouldn't assume they know that.